src/algoesup/test.py

```diff
@@ -3,6 +3,24 @@
 from typing import Callable
 
 
+def validate_test_table(function: Callable, test_table: list) -> list:
+    """Validate the structure of a test table"""
+    errors = []
+    if not isinstance(test_table, (list, tuple)):
+        errors.append("The test table must be a list or tuple.")
+        return errors
+    # The number of arguments function expects.
+    expects = function.__code__.co_argcount
+    for num, test_case in enumerate(test_table, start=1):
+        if not isinstance(test_case, (list, tuple)):
+            errors.append(f"Test case {num}: should be a list or tuple.")
+        elif not isinstance(test_case[0], str):
+            errors.append(f"Test case {num}: first element must be a string.")
+        elif len(test_case[1:-1]) != expects:
+            num_args = len(test_case[1:-1])
+            errors.append(f"Test case {num}: {num_args} inputs found instead of {expects}")
```

Comment:
Here you can use the test case name because the first arg is a string.

Reply:
Good idea, thanks.